lib/Interpreter/ValueExtractionSynthesizer.cpp

```diff
@@ -167,13 +167,19 @@ namespace cling {
         Expr* SVRInit = SynthesizeSVRInit(lastExpr);
         // if we had return stmt update to execute the SVR init, even if the
         // wrapper returns void.
-        if (RS) {
-          if (ImplicitCastExpr* VoidCast
-              = dyn_cast<ImplicitCastExpr>(RS->getRetValue()))
-            VoidCast->setSubExpr(SVRInit);
+        if (SVRInit) {
+          if (RS) {
+            if (ImplicitCastExpr* VoidCast
+                = dyn_cast<ImplicitCastExpr>(RS->getRetValue()))
+              VoidCast->setSubExpr(SVRInit);
+          }
+          else
+            **I = SVRInit;
+        }
+        else {
+          // FIXME: Is this leaking anything (lastExpr?)
```

Comment:
Is this more descriptive:
```
// FIXME: Do this atomically or something so that AST context will not
// contain Expr(s) that are unused for the rest of it's life.
```